src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -886,6 +886,48 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
     };
 #endif
 
+    template<typename T>
+    void SetValueForKey(Json::Value& json, std::string_view key, const T& target);
+
+    template<typename T>
+    std::decay_t<T> GetValue(const Json::Value& json);
+
+    template<>
+    struct ConversionTrait<winrt::Windows::Foundation::Collections::StringMap>
+    {
+        winrt::Windows::Foundation::Collections::StringMap FromJson(const Json::Value& json)
+        {
+            winrt::Windows::Foundation::Collections::StringMap stringMap{};
+            for (Json::Value::const_iterator it = json.begin(); it != json.end(); ++it)
+            {
+                stringMap.Insert(GetValue<winrt::hstring>(it.key()), GetValue<winrt::hstring>(*it));
+            }
+            return stringMap;
+        }
+
+        bool CanConvert(const Json::Value&)
+        {
+            return true;

```

Comment:
we may want to make sure that the json thing is an object

Reply:
technically we could also spend time making sure that Trait<string>.CanConvert() each value, but... eh. It'll probably throw during conversion on 903 anyway and be OK